### PR TITLE
Improve create process from template

### PR DESCRIPTION
**Pull request by IkramMaalej**

### CODE REVIEW

File: Kitodo/src/main/java/org/kitodo/production/forms/createprocess/CreateProcessForm.java [hunk 1]
```diff
@@ -89,6 +91,30 @@ public RulesetManagementInterface getRuleset() {
         return rulesetManagementInterface;
     }
 
+    public void updateRulesetAndDocType(String rulesetFileName) {
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/production/forms/createprocess/CreateProcessForm.java]: Missing javadoc

File: Kitodo/src/main/java/org/kitodo/production/forms/createprocess/CreateProcessForm.java [hunk 2]
```diff
@@ -478,4 +478,16 @@ private RulesetManagementInterface openRulesetFile(String fileName) throws IOExc
         }
         return ruleset;
     }
+
+    public void initializePocesses() {
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/production/forms/createprocess/CreateProcessForm.java]: missing javadoc

File: Kitodo/src/main/java/org/kitodo/production/forms/createprocess/CreateProcessForm.java [hunk 3]
```diff
@@ -478,4 +478,16 @@ private RulesetManagementInterface openRulesetFile(String fileName) throws IOExc
...
+        try {
+            ProcessGenerator processGenerator = new ProcessGenerator();
+            processGenerator.generateProcess(template.getId(), project.getId());
+            this.processes = new LinkedList<>(Collections.singletonList(processGenerator.getGeneratedProcess()));
+        } catch (ProcessGenerationException e) {
+            logger.error(e.getLocalizedMessage());
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/production/forms/createprocess/CreateProcessForm.java]: Error should be transfered to frontend, shouldn't it?

File: Kitodo/src/main/java/org/kitodo/production/forms/createprocess/ProcessMetadataTab.java [hunk 4]
```diff
@@ -131,7 +131,7 @@ void fillProcessDetailsElements(List<ProcessDetail> processDetailList, NodeList
             boolean filled = false;
             for (ProcessDetail detail : processDetailList) {
                 if (Objects.nonNull(detail.getMetadataID()) && detail.getMetadataID().equals(nodeName)) {
-                    if (isChild || getProcessDetailValue(detail).isEmpty()) {
+                    if (isChild || Objects.isNull(getProcessDetailValue(detail)) || getProcessDetailValue(detail).isEmpty()) {
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/production/forms/createprocess/ProcessMetadataTab.java]: use StringUtils.isBlank()  (this is checking for null and empty)

File: Kitodo/src/main/resources/messages/errors_de.properties [hunk 5]
```diff
@@ -75,6 +75,7 @@ invalidIdentifierSame=Das Identifikationsmerkmal {0} enth\u00E4lt in {1} und {2}
 # K
 
 # L
+errorLoadingDocTypes=Regelsatz Konfiguration Fehler: Kein DocType ('division') gefunden
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/resources/messages/errors_de.properties]:
```suggestion
errorLoadingDocTypes=Regelsatz Konfigurationsfehler: Kein DocType ('division') gefunden
```

File: Kitodo/src/main/java/org/kitodo/production/forms/createprocess/CreateProcessForm.java (hunk 3 above)
Comment (IkramMaalej) [reply Kitodo/src/main/java/org/kitodo/production/forms/createprocess/CreateProcessForm.java]: In this case, there is no need to transfer the error to frontend.

**Pull request merged**